File: sentry-android-core/src/main/java/io/sentry/android/core/MainLooperHandler.java
package io.sentry.android.core;

import android.os.Handler;
import android.os.Looper;
import org.jetbrains.annotations.NotNull;

final class MainLooperHandler {
  private final @NotNull Handler handler;

  MainLooperHandler() {
    this(Looper.getMainLooper());
  }

  MainLooperHandler(final @NotNull Looper looper) {
    handler = new Handler(looper);
  }

  public void post(final @NotNull Runnable runnable) {
    handler.post(runnable);
  }

  public @NotNull Thread getThread() {
    return handler.getLooper().getThread();
  }
}

File: sentry-android-replay/src/main/java/io/sentry/android/replay/util/MainLooperHandler.kt
package io.sentry.android.replay.util

import android.os.Handler
import android.os.Looper

internal class MainLooperHandler(looper: Looper = Looper.getMainLooper()) {
  val handler = Handler(looper)

  fun post(runnable: Runnable): Boolean {
    return handler.post(runnable)
  }

  fun postDelayed(runnable: Runnable?, delay: Long): Boolean {
    return handler.postDelayed(runnable ?: return false, delay)
  }

  fun removeCallbacks(runnable: Runnable?) {
    handler.removeCallbacks(runnable ?: return)
  }
}
